### PR TITLE
Add setting for the main apiserver hostname

### PR DESCRIPTION
**Pull request by ereslibre**
Fixes bsc#1039437

### CODE REVIEW

File: lib/velum/kubernetes.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def self.kubeconfig
                                           targets:     "roles:kube-master",
                                           target_type: "grain",
                                           arg:         "cat /etc/pki/minion.key"
-      host = Minion.master.applied.pluck(:fqdn).first
+      host = Pillar.value(pillar: :apiserver) || Minion.master.applied.pluck(:fqdn).first
```
Comment (kiall) [inline lib/velum/kubernetes.rb]: It looks like we've added the new field as required, so the pillar value should always exist. Should we remove the `|| Minion.master.applied.pluck(:fqdn).first`?

File: app/views/setup/welcome.html.slim [hunk 2]
```diff
@@ -9,7 +9,10 @@ h1 Initial CaaSP Configuration
 = form_for :settings, url: setup_path, method: :put do |f|
   .form-group
     = f.label :dashboard, "Dashboard location"
-    = f.text_field :dashboard, value: (Pillar.value(pillar: :dashboard) || request.host), class: "form-control"
+    = f.text_field :dashboard, value: (Pillar.value(pillar: :dashboard) || request.host), class: "form-control", required: true
+  .form-group
+    = f.label :apiserver, "External Kubernetes API server hostname"
```
Comment (kiall) [inline app/views/setup/welcome.html.slim]: FQDN is likely better than hostname, I know, I know - this name was my suggestion, my bad! A hostname is `foo`, a FQDN is `foo.mycompany,com`.

Comment (ereslibre) [inline app/views/setup/welcome.html.slim]: Yeah... but technically anything the customer set as resolvable on his/her side will work (whether it's a hostname or a fqdn). I am OK with showing fqdn here though.

File: lib/velum/kubernetes.rb (hunk 1 above)
Comment (ereslibre) [inline lib/velum/kubernetes.rb]: Yes, I will remove it, it was to be extra safe here, but it doesn't make sense if we require it already.

**Pull request merged**